src/parse/parse_step.go

```diff
@@ -64,6 +64,27 @@ func parse(tid int, state *core.BuildState, label, dependent core.BuildLabel, fo
 		if err := subrepo.LoadSubrepoConfig(); err != nil {
 			return err
 		}
+
+		// Validate plugin ID is the same as the subrepo name
+		if pluginID := state.RepoConfig.PluginDefinition.Name; pluginID != "" {
+			if !strings.EqualFold(pluginID, subrepo.Name) {
+				log.Errorf("Subrepo name \"%s\" should be the same as the plugin ID \"%s\"", subrepo.Name, pluginID)
+			}
+		}
+
+		// Validate the plugin config keys set in the host repo
+		definedKeys := map[string]bool{}
+		for key, definition := range state.RepoConfig.PluginConfig {
+			configKey := getConfigKey(key, definition.ConfigKey)
+			definedKeys[configKey] = true
+		}
+		if plugin := state.Config.Plugin[subrepo.Name]; plugin != nil {
+			for key := range plugin.ExtraValues {
+				if _, ok := definedKeys[strings.ToLower(key)]; !ok {
+					log.Warning("Unrecognised config key \"%v\" for plugin \"%v\"", key, subrepo.Name)
```

Comment:
Want to have a chat about making these mandatory too? I think Please only makes them warnings so it's able to do basic commands with invalid config. For plugins, it's far less problematic so we can probably return an error here too.

Reply:
Yeah makes sense!